src/main/java/com/google/step/YOUR_PROJECT_NAME_HERE/external/StackOverflowClient.java

```diff
@@ -0,0 +1,152 @@
+  private static final String QUESTION_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/questions/%s/answers?"
+          + "order=desc&sort=votes&site=stackoverflow";
+  // This url specify filter to generate answer body.
+  private static final String ANSWER_URL_TEMPLATE =
+      "https://api.stackexchange.com/2.2/answers/%s?order"
+          + "=desc&sort=activity&site=stackoverflow&filter=!9_bDE(fI5";
+  private static final int ID_INDEX = 2;
+  private static final String ITEM_PARAMETER = "items";
+  private static final String TITLE_PARAMETER = "title";
+  private static final String BODY_PARAMETER = "body";
+  private static final String CODE_PARAMETER = "code";
+  private static final String ANSWER_ID_PARAMETER = "answer_id";
+  // Set 200 to be the maximum length of description for MVP.
+  private static final int DESCRIPTION_LENGTH_PARAMETER = 200;
+
+  /* This method build the desired card after getting each field. */
+  /* It will return a null card if no valid question is found */
+  public Card search(String url) {
+    String questionId;
+    try {
+      questionId = getQuestionId(url);
+    } catch (URISyntaxException e) {
+      // Return null card if no valid card available.
+      return null;
+    }
+    String answerId = getAnswerId(questionId);
+    if (answerId == null) {
+      return null;
+    }
+    String title = getTitle(questionId);
+    String answerBody = getAnswerBody(answerId);
+    String description = getDescription(answerBody);
+    String code = getCode(answerBody);
+    Card card = new Card(title, code, url, description);
+    return card;
+  }
+
+  /* Get the question id of passed in URL. */
+  private String getQuestionId(String url) throws URISyntaxException {
+    URI uri = new URI(url);
+    // Parse the URL to get the question id.
+    String[] segments = uri.getPath().split("/");
+    String questionId = segments[ID_INDEX];
+    if (!Pattern.matches("[0-9]+", questionId)) {
+      return null;
+    }
+    return questionId;
+  }
+
+  /* Get the question title using question id */
+  private String getTitle(String questionId) {
+    String searchUrl = String.format(SEARCH_URL_TEMPLATE, questionId);
+    String title = getResponse(searchUrl, TITLE_PARAMETER);
+    return title;
+  }
+
+  /* Get the most voted answer's id and store it in the card. */
+  private String getAnswerId(String questionId) {
+    String questionUrl = String.format(QUESTION_URL_TEMPLATE, questionId);
+    String answerId = getResponse(questionUrl, ANSWER_ID_PARAMETER);
+    // Replace the question id by the answer id in order to retrieve the code body next.
+    return answerId;
+  }
+
+  /* Get the content of the answer and store it in the card. */
+  private String getAnswerBody(String answerId) {
+    String answerUrl = String.format(ANSWER_URL_TEMPLATE, answerId);
+    String body = getResponse(answerUrl, BODY_PARAMETER);
+    return body;
+  }
+
+  /* Get the description using return answer body. */
```

Comment:
Nit: instead of `return answer body` maybe change to `Return the description parsed from the answer body.`